avm/ptn/app/paas-ase-cosmosdb-tier4/main.bicep

```
@@ -0,0 +1,365 @@
+metadata name = 'PaaS ASE with CosmosDB Tier 4'
+metadata description = 'Creates a PaaS ASE with CosmosDB Tier 4 resiliency configuration.'
+
+// ================ //
+// Parameters       //
+// ================ //
+// @description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+// param enableDefaultTelemetry bool = true
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Required. The name of the deployment.')
+param name string
+
+@description('Optional. Suffix for all resources.')
+param suffix string = ''
+
+// Common Parameters
+
+@description('Optional. Tags of the resource.')
+param tags object = {}
```

Not yet enforced, but eventually for v1.0 all objects/arrays of objects must be typed. Ref [BCPNFR1](https://azure.github.io/Azure-Verified-Modules/spec/BCPNFR1/)

Tags are here used for multiple referenced resources, but their type is the same. Hence suggesting resource derived types of the first resource module referenced in the list

```suggestion
param tags resourceInput<'Microsoft.Network/networkSecurityGroups@2024-07-01'>.tags?
```